### ISSUE TITLE
pose control

### ISSUE THREAD

**Issue by taotaoorange** [HaiyuWu/Vec2Face#6]
Thanks for sharing the data, code and models. 
It looks like in the current implementation, only 'yaw' is considered for pose control. What about 'pitch' and 'roll'? 

Can the learned model extrapolate?  If the training data for the generative model only cover a limited range of head pose, can it generate face images beyond the range of the head pose in the training data?

**Comment by HaiyuWu** [comment]
> It looks like in the current implementation, only 'yaw' is considered for pose control. What about 'pitch' and 'roll'?
>
> Can the learned model extrapolate? If the training data for the generative model only cover a limited range of head pose, can it generate face images beyond the range of the head pose in the training data?

Since "roll" is solved by face detectors, it is not necessary to generate different roll angles. As for pitch, Vec2Face definitely can generate different pitch angles and I have done this experiment. However, I found that pitch angle did not help the accuracy on the test set a lot, so I just did pose control on "yaw".

As for model extrapolate, it is an really interesting question. I don't know the exact answer for it. I would say large pitch angles and yaw angles rarely appear in the dataset in terms of the dataset size, and Vec2Face can somehow generate these types of variations, so it has generalizability on minority attributes but I am not sure for the attributes that it has never seen.

**Comment by taotaoorange** [comment]
Thanks for your quick reply. To try out "pitch", is it only about changing this line of code "yaw_loss = torch.abs(pose - torch.abs(pose_info[:, 1].clip(min=-90, max=90)))"

**Comment by HaiyuWu** [comment]
Yes.

**Comment by yeluohuazhifou** [comment]
> Thanks for your quick reply. To try out "pitch", is it only about changing this line of code "yaw_loss = torch.abs(pose - torch.abs(pose_info[:, 1].clip(min=-90, max=90)))"

change to "pitch_loss = torch.abs(pose - torch.abs(pose_info[:, 0].clip(min=-90, max=90)))" is for pitch control? or "torch.abs(pose - torch.abs(pose_info[:, 2].clip(min=-90, max=90)))" ?

**Comment by HaiyuWu** [comment]
> change to "pitch_loss = torch.abs(pose - torch.abs(pose_info[:, 0].clip(min=-90, max=90)))" is for pitch control? or "torch.abs(pose - torch.abs(pose_info[:, 2].clip(min=-90, max=90)))" ?

If you are using the provided pose estimator, the prediction order for each image is ["pitch, yaw, roll"](https://github.com/thohemp/6DRepNet?tab=readme-ov-file#pip-install). By the way, the pitch range is -45 to 45.
